src/vs/workbench/contrib/chat/common/promptFileReference.ts

```diff
@@ -123,6 +122,14 @@ export class PromptFileReference extends Disposable {
 		}
 	}
 
+	/**
+	 * Subscribe to the `onUpdate` event.
+	 * @param callback
+	 */
+	public onUpdate(callback: () => unknown) {
+		this._register(this._onUpdate.event(callback));
```

Comment:
I'd say that in principle, the code attaching the listener should control the lifecycle of that listener. However this problem of wanting a simple listener that just tracks the lifecycle of the object comes up really often. I'm fine with this pattern, I did something similar here: https://github.com/microsoft/vscode/blob/roblou/thick-turtle/src/vs/workbench/contrib/chat/browser/chatContentParts/chatTreeContentPart.ts#L87

There's probably a larger conversation we could have in slack with the team about what sort of pattern we should prefer for handling disposables like this